src/transcoding.ts

```diff
@@ -257,10 +260,50 @@ export function requestChangeCaseAndCleanup(
   return convertedRequest;
 }
 
+export function isProto3OptionalField(field: Field) {
+  return field && field.options && field.options![proto3OptionalName];
+}
+
+export function isRequiredField(field: Field) {
+  return (
+    field &&
+    field.options &&
+    field.options![fieldBehaviorOptionName] === 'REQUIRED'
+  );
+}
+
+export function getFieldNameOnBehavior(
```

Comment:
nit: the name somewhat confuses me, though I have a hard time coming up with a better name. Maybe just `getFieldNamesByBehavior`?